clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -136,6 +137,22 @@ public void close() throws IOException {
       }
     }
 
+    canComplete = canComplete || storeToTachyon();
+
+    if (canComplete) {
+      FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+      try {
+        masterClient.completeFile(mFileId);
+      } catch (TException e) {
+        throw new IOException(e);
+      } finally {
+        mContext.releaseMasterClient(masterClient);
+      }
+    }
+    mClosed = true;
+  }
+
+  protected boolean storeToTachyon() throws IOException {
```

Comment:
the name is not descriptive at all, because if `mCanceled` is true, all bocks are canceled.